### DISCUSSION TITLE
Post vs Deposit and Bulk Outbox Inserts

### DISCUSSION THREAD

**Discussion by preardon** [BrighterCommand/Brighter#1581] [Q&A] [answered]
Hey Guys, I'm going to go over the code and look at a few things to really round off my thoughts here. Is there any reason we wouldn't want to be able to bulk Post to the outbox.

Am I correct in thinking that the Point of Deposit Post is so that you can Pre-load messages in the outbox before Dispatch?
And if this is the case is there any reason why we wouldn't  want a Bulk Deposit?

**Comment by iancooper** [comment]
I know we have had this conversation now, so you know about the intent of DepositPost, but just for anyone else.

Deposit is intended for use within a database transaction that writes both entities to the backing store and any associated events as part of the Outbox pattern: https://www.kamilgrzybek.com/design/the-outbox-pattern/

You can then either explicitly Clear the Outbox to send any messages without delay, or use a Sweeper to flush any unsent messages on a timed basis.

In principle we could write a whole set to DepositPost, if your changes required multiple downstream messages. Normally, I'd just iteratively add, but I don't see any reason why we couldn't have a bulk version if it was desirable.

**Reply by preardon** [reply]
Thanks Ian, I think it could be nice to have, but my issues were more around Batch processing, Maybe a nice feature to add for the future but not urgent

**Comment by preardon** [comment, marked as answer]
This has been resolved in v9.1.x command there is now a bulk async message producer, and an implicit clear Async on the command processor that has a bulk flag (also plumbed through to the sweeper)

**Reply by nesbo** [reply]
thanks mate, but, believe it or not, I need one synchronous version of it :) I know what I need to make then.

**Reply by preardon** [reply]
> nesbo

I can work on this if you would like, however it is important to note that at the minute we only support bulk on the implicit clear (sweeping) path

The reason that I didn't do this Sync was as I was using IAsyncEnumumerable, however we can code around this if this is what you need, however I am interested to hear why you want a Sync version as in the use cases that I've got these run in a separate process.  Lets have a chat when you get a minute and come up with a solution.

Hope you're enjoying the new Job mate :)

**Reply by nesbo** [reply]
yeah, depositing to outbox is what I need exactly.

I am building synchronous pipes that read raw byte streams from TCP sockets open to some distant devices. I already have concurrency in the process, with my threads per socket/device, so I don't want more context switching. also, I am responsible to talk to the device synchronously, aka ack when I process everything I received. I'll get back to this soon.

**Reply by preardon** [reply]
Ah, we haven't done Bulk Inserts as yet, only bulk publish
